spec/support/helpers.rb

```diff
@@ -22,4 +22,60 @@ def destroy_repos
   def extract_text(string)
     Nokogiri::HTML(string).text.strip.gsub(/\s+/, ' ')
   end
+
+  def submit_files_before_due_date
+    pretend_now_is(Time.parse('July 20 2009 5:00PM')) do
+      # expect(Time.now).to be < @assignment.due_date
+      # expect(Time.now).to be < @assignment.submission_rule.calculate_collection_time
+      # expect(Time.now).to be < @assignment.submission_rule.calculate_grouping_collection_time(@membership.grouping)
+
+      @group.access_repo do |repo|
+        txn = repo.get_transaction('test')
+        txn = add_file_helper(@assignment, txn, 'TestFile.java', 'Some contents for TestFile.java')
```

Comment:
Same thing here... no hardcoded file names and also instance variables (like `@assignment` and `@group` should not be used in helper methods. This is because, this helper method could be called from any spec and there is no guarantee that this instance variable would necessarily be defined.